test/e2e/placement_drift_diff_test.go

```diff
@@ -0,0 +1,2295 @@
+				*expectedOwnerRef,
+			}
+
+			// No need to use an Eventually block as this spec runs after the CRP status has been verified.
+			diff = cmp.Diff(
+				cm, &wantCM,
+				ignoreObjectMetaAutoGenExceptOwnerRefFields,
+				ignoreObjectMetaAnnotationField,
+			)
+			Expect(diff).To(BeEmpty(), "ConfigMap diff (-got +want):\n%s", diff)
+		})
+
+		It("should place the resources on the rest of the member clusters", func() {
+			targetClusters := []*framework.Cluster{
+				memberCluster2EastCanary,
+				memberCluster3WestProd,
+			}
+			for idx := range targetClusters {
+				memberCluster := targetClusters[idx]
+
+				workResourcesPlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(memberCluster)
+				Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work resources on member cluster %s", memberCluster.ClusterName)
+			}
+		})
+
+		AfterAll(func() {
+			ensureCRPAndRelatedResourcesDeleted(crpName, allMemberClusters)
+		})
+	})
+})
+
+var _ = Describe("report diff mode", func() {
+	Context("do not touch anything", Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+		nsName := fmt.Sprintf(workNamespaceNameTemplate, GinkgoParallelProcess())
+		cmName := fmt.Sprintf(appConfigMapNameTemplate, GinkgoParallelProcess())
+
+		BeforeAll(func() {
+			// Create the resources on the hub cluster.
+			createWorkResources()
+
+			// Create the resources on one of the member clusters.
+			ns := appNamespace()
+			// Add a label (unmanaged field) to the namespace.
+			ns.Labels = map[string]string{
+				unmanagedLabelKey:      unmanagedLabelVal1,
+				workNamespaceLabelName: fmt.Sprintf("%d", GinkgoParallelProcess()),
+			}
+			Expect(memberCluster1EastProdClient.Create(ctx, &ns)).To(Succeed())
+
+			cm := appConfigMap()
+			// Update the configMap data (managed field).
+			cm.Data = map[string]string{
+				managedDataFieldKey: managedDataFieldVal1,
+			}
+			Expect(memberCluster1EastProdClient.Create(ctx, &cm)).To(Succeed())
+
+			crp := &placementv1beta1.ClusterResourcePlacement{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: crpName,
+					// Add a custom finalizer; this would allow us to better observe
+					// the behavior of the controllers.
+					Finalizers: []string{customDeletionBlockerFinalizer},
+				},
+				Spec: placementv1beta1.ClusterResourcePlacementSpec{
+					ResourceSelectors: workResourceSelector(),
+					Strategy: placementv1beta1.RolloutStrategy{
+						Type: placementv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+							UnavailablePeriodSeconds: ptr.To(2),
+						},
+						ApplyStrategy: &placementv1beta1.ApplyStrategy{
+							ComparisonOption: placementv1beta1.ComparisonOptionTypeFullComparison,
+							Type:             placementv1beta1.ApplyStrategyTypeReportDiff,
+							WhenToTakeOver:   placementv1beta1.WhenToTakeOverTypeNever,
```

Comment:
what happens if this is not never?

Reply:
Ah, the resources will be taken over, but there would be no diff reported. When the CRP is deleted the resources will be removed.